### ISSUE TITLE
Inconsistent predicted results compared to the example

### ISSUE THREAD

**Issue by rosa840905** [kexinhuang12345/DeepPurpose#93]
Thank you so much for your great work.
 I ran the below code from oneliner_repurpose_LCK_gene but got different predicted results compared to your example.
(I didn't adjust any parameters)
<code>oneliner.repurpose(target = target, 
                    target_name = target_name, 
                    save_dir = './save_folder',
                    pretrained_dir = './save_folder/pretrained_models/DeepPurpose_BindingDB/')</code>
My result is below figure:
![image](https://user-images.githubusercontent.com/53463968/110623407-c0f27900-81d7-11eb-9f27-47988edd23cd.png)

Your example:
![image](https://user-images.githubusercontent.com/53463968/110623370-b6d07a80-81d7-11eb-8728-5a5c163f138c.png)

Why I got different predicted results compared to your example, please?
Thank you.

**Comment by kexinhuang12345** [comment]
Resolved through emails. Closing here.

**Comment by chemlove** [comment]
I am getting the same problem. How did you solve this?

Another question: 
The lines 326-331 in oneliner.py:
```
	with open(os.path.join(result_folder_path, 'logits_VS_mean.pkl'), 'wb') as f:
		pickle.dump(np.mean(y_preds_models, axis = 0), f, pickle.HIGHEST_PROTOCOL) 
	with open(os.path.join(result_folder_path, 'logits_VS_max.pkl'), 'wb') as f:
		pickle.dump(np.min(y_preds_models, axis = 0), f, pickle.HIGHEST_PROTOCOL) 
	with open(os.path.join(result_folder_path, 'logits_VS_mean_max.pkl'), 'wb') as f:
		pickle.dump((np.min(y_preds_models, axis = 0) + np.mean(y_preds_models, axis = 0))/2, f, pickle.HIGHEST_PROTOCOL) 
```
Should the convert_y option be considered when saving the logits_VS_max.pkl and logits_VS_mean_max.pkl?

**Comment by chemlove** [comment]
Kexin, I tried the `DEMO/Make-DAVIS-Correlation-Figure.ipynb` using the pre-trained BindingDB Kd models to predict the affinity of DAVIS dataset and to check the accuracy of DeepPurpose.

With your pre-trained BindingDB Kd models, the output Pearson correlation coefficients of several random 1000 drug-target paris were always under 0.05. Howerver, the correlation coefficient shown in you notebook is 0.7789. 

My pre-trained BindingDB Kd models were downloaded using `download_pretrained_model(model_name)`. Are these models the same with those used in your demo notebook?

**Comment by kexinhuang12345** [comment]
Hi, did you use the oneliner mode? because it aggregates five models. One model may not be sufficient.

**Comment by chemlove** [comment]
> Hi, did you use the oneliner mode? because it aggregates five models. One model may not be sufficient.

I used the oneliner mode in which all five models were employed. And I also checked the output of each model. The results were so weird that I think the pre-trained models I used were not those you had used in your demo notebook.

My codes and results:

```
import os
#os.chdir('../')
from DeepPurpose import dataset
from DeepPurpose import oneliner
from DeepPurpose.utils import convert_y_unit

import numpy as np
import pandas as pd
import pickle
from scipy.stats import pearsonr

SMILES, Target_seq, y = dataset.load_process_DAVIS()

df_data = pd.DataFrame(zip(SMILES, Target_seq, y))
df_data.rename(columns={0:'SMILES',
                        1: 'Target Sequence',
                        2: 'Label'}, 
                        inplace=True)

df_1000 = df_data.sample(n = 1000, replace = False).reset_index(drop = True)

oneliner.virtual_screening(target = df_1000['Target Sequence'].values, target_name = None, 
                X_repurpose = df_1000['SMILES'].values, 
                drug_names = None,
                save_dir = './save_folder',               
                pretrained_dir = './model_download/pretrained_models/',   # downloaded using download_pretrained_model(model_name) for each model
                convert_y = True,
                subsample_frac = 1,
                pretrained = True,
                split = 'random',
                frac = [0.7,0.1,0.2],
                agg = 'agg_mean_max',
                output_len = 30)

with open('./save_folder/results_aggregation/logits_VS_mean_max.pkl', 'rb') as f:
    list_ = pickle.load(f)
result = convert_y_unit(list_, 'nM', 'p')

pearsonr(result, df_1000.Label.values)

# output:  (-0.007272592358385338, 0.8183288107558107)
```

**Comment by chemlove** [comment]
>> The lines 326-331 in oneliner.py:
```
with open(os.path.join(result_folder_path, 'logits_VS_mean.pkl'), 'wb') as f:
    pickle.dump(np.mean(y_preds_models, axis = 0), f, pickle.HIGHEST_PROTOCOL) 
with open(os.path.join(result_folder_path, 'logits_VS_max.pkl'), 'wb') as f:
    pickle.dump(np.min(y_preds_models, axis = 0), f, pickle.HIGHEST_PROTOCOL) 
with open(os.path.join(result_folder_path, 'logits_VS_mean_max.pkl'), 'wb') as f:
    pickle.dump((np.min(y_preds_models, axis = 0) + np.mean(y_preds_models, axis = 0))/2, f, pickle.HIGHEST_PROTOCOL) 
```
>> Should the convert_y option be considered when saving the logits_VS_max.pkl and logits_VS_mean_max.pkl?

> Regarding the first question, yes, you can convert it after as well. This is just to save the raw scores.

Lines 313-324 of oneliner.py:
```
if agg == 'mean':
    y_pred = np.mean(y_preds_models, axis = 0)
elif agg == 'max_effect':
    if convert_y:
        y_pred = np.min(y_preds_models, axis = 0)
    else:
        y_pred = np.max(y_preds_models, axis = 0)
elif agg == 'agg_mean_max':
    if convert_y:
        y_pred = (np.min(y_preds_models, axis = 0) + np.mean(y_preds_models, axis = 0))/2
    else:
        y_pred = (np.max(y_preds_models, axis = 0) + np.mean(y_preds_models, axis = 0))/2
```

>> Should the convert_y option be considered when saving the logits_VS_max.pkl and logits_VS_mean_max.pkl?

I think lines 326-331 should also consider the convert_y option as in lines 313-324. The logits_VS_max.pkl file would be correct only if convert_y == True since np.min() is used.

**Comment by kexinhuang12345** [comment]
Hi, i think it is due to different pretraining model. Can you turn "pretrained_dir = None"? it will then download the five pretrained model used in creating the figure. The individual model you use maybe bindingdb ic50? since davis is Kd, this might be the reason why the transferrablity is bad.

**Comment by kexinhuang12345** [comment]
closing for now; if the issue continues, feel free to reopen
